pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java

```diff
@@ -39,69 +41,81 @@
 
     @ColumnName(DBConstants.DELETED_AT)
-    private final long deletedAt;
+    private Long deletedAt;
 
     @ColumnName(DBConstants.IS_VIRUS_FREE)
     private Boolean isVirusFree;
 
+    @ColumnName(DBConstants.SENT_AT)
+    private Long sentAt;
+
     private static final String SQL_SELECT_DOCUMENTS_BASE = "SELECT sD.id, p.ddp_instance_id, "
             + "p.ddp_participant_id, p.participant_id, sD.file_name, sD.mime_type, sD.bucket, sD.blob_path, "
-            + "sD.created_by_user_id, sD.created_at, sD.deleted_by_user_id, sD.deleted_at, sD.is_virus_free "
+            + "sD.created_by_user_id, sD.created_at, sD.deleted_by_user_id, sD.deleted_at, sD.is_virus_free, t.created_date  "
             + "FROM  somatic_documents sD "
             + "LEFT JOIN ddp_instance as ddp on (ddp.ddp_instance_id = sD.ddp_instance_id) "
-            + "LEFT JOIN ddp_participant p on (p.participant_id = sD.participant_id) ";
+            + "LEFT JOIN ddp_participant p on (p.participant_id = sD.participant_id) "
+            + "LEFT JOIN ddp_survey_trigger t on (t.survey_trigger_id = sD.trigger_id) "
+    ;
 
     private static final String SQL_SELECT_DOCUMENTS_BY_REALM = SQL_SELECT_DOCUMENTS_BASE
             + "WHERE ddp.instance_name = ?";
 
     private static final String SQL_SELECT_DOCUMENTS_BY_REALM_AND_PTPT = SQL_SELECT_DOCUMENTS_BASE
             + "WHERE ddp.instance_name = ? AND p.ddp_participant_id = ?";
 
+    private static final String SQL_SELECT_DOCUMENT_BY_REALM_PTPT_DOCID = SQL_SELECT_DOCUMENTS_BY_REALM_AND_PTPT
+            + " AND sD.id = ?";
+
     private static final String SQL_SELECT_DOCUMENT_BY_ID_AND_REALM = SQL_SELECT_DOCUMENTS_BASE
             + "WHERE sD.id = ? AND ddp.instance_name = ?";
 
     private static final String SQL_SELECT_DOCUMENT_BY_BUCKET_AND_PATH = SQL_SELECT_DOCUMENTS_BASE
             + "WHERE sD.bucket = ? AND sD.blob_path = ?";
 
+    private static final String SQL_UPDATE_SENT_AT_TIME = "UPDATE somatic_documents AS sD "
+            + "LEFT JOIN ddp_instance as ddp ON (ddp.ddp_instance_id = sD.ddp_instance_id) SET sD.trigger_id = ? "
+            + "WHERE sD.id = ? AND ddp.instance_name = ?";
+
     private static final String SQL_UPDATE_VIRUS_STATUS_SUCCESS = "UPDATE somatic_documents SET is_virus_free = ?, "
             + "bucket = ?, blob_path = ? WHERE bucket = ? AND blob_path = ?";
 
     private static final String SQL_UPDATE_VIRUS_STATUS_FAILED = "UPDATE somatic_documents SET is_virus_free = ?, "
             + "deleted_at = ? WHERE bucket = ? AND blob_path = ?";
 
-    private static final String SQL_DELETE_DOCUMENT_BY_DOCUMENT_ID_AND_REALM = "UPDATE somatic_documents "
-            + "LEFT JOIN ddp_instance as ddp ON (ddp.ddp_instance_id = ddp_instance_id) SET deleted_by_user_id = ?, "
-            + "deleted_at = ? WHERE id = ? AND ddp.instance_name = ?";
+    private static final String SQL_DELETE_DOCUMENT_BY_DOCUMENT_ID_AND_REALM = "UPDATE somatic_documents AS sD "
```

Comment:
For some reason, this change was undone before I committed.  It's back and has been manually verified as working.